### PR TITLE
lightning: add foreign key test

### LINKED ISSUE #40226: add foreign key lightning test

### PR DESCRIPTION
**Pull request by crazycs520**
Signed-off-by: crazycs520 <crazycs520@gmail.com>

### What problem does this PR solve?

Issue Number: close #40226

Problem Summary: add foreign key lightning test. 

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by gozssky — APPROVED** [review]

File: br/tests/lightning_foreign_key/run.sh [hunk 1]
```diff
@@ -16,13 +16,21 @@
...
 set -eu
 
-# Create existing tables that import data will reference.
+run_sql 'DROP DATABASE IF EXISTS fk;'
 run_sql 'CREATE DATABASE IF NOT EXISTS fk;'
+# Create existing tables that import data will reference.
 run_sql 'CREATE TABLE fk.t2 (a BIGINT PRIMARY KEY);'
 
 for BACKEND in tidb local; do
-  run_sql 'DROP TABLE IF EXISTS fk.t;'
```
Comment (okJiang) [inline br/tests/lightning_foreign_key/run.sh:24]: due to delete this line, so the `local` backend test failed.

**Review by lichunzhu — APPROVED** [review]
LGTM

**Review by okJiang — APPROVED** [review]

**Pull request merged**